pkg/kube/multicluster/secretcontroller.go

```diff
@@ -91,15 +92,36 @@ type Controller struct {
 
 // NewController returns a new secret controller
 func NewController(kubeclientset kube.Client, namespace string, localClusterID cluster.ID) *Controller {
+	// When these two are set to true, Istiod will be watching the namespace in which
+	// Istiod is running on the external cluster. Use the inCluster credentials to
+	// create a kubeclientset
+	informerClient := kubeclientset
+	if features.LocalClusterSecretWatcher && features.ExternalIstiod {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			log.Errorf("Could not get istiod incluster configuration: %v", err)
+			return nil
+		}
+		log.Info("Successfully retrieved incluster config.")
+
+		localKubeClient, err := kube.NewClient(kube.NewClientConfigForRestConfig(config))
+		if err != nil {
+			log.Errorf("Could not create a client to access local cluster API server: %v", err)
+			return nil
+		}
+		log.Infof("Successfully created in cluster kubeclient at %s", localKubeClient.RESTConfig().Host)
+		informerClient = localKubeClient
+	}
+
 	secretsInformer := cache.NewSharedIndexInformer(
 		&cache.ListWatch{
 			ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
 				opts.LabelSelector = MultiClusterSecretLabel + "=true"
-				return kubeclientset.Kube().CoreV1().Secrets(namespace).List(context.TODO(), opts)
+				return informerClient.Kube().CoreV1().Secrets(namespace).List(context.TODO(), opts)
```

Comment:
So we want to stop reading from the config cluster ? I thought you want to read from both, it may break some APIs or apps if we stop reading from the config cluster.

Reply:
It is controlled by the totally new `env`